beefy-gadget/src/keystore.rs

```diff
@@ -54,11 +54,10 @@ impl BeefyKeystore {
 	///
 	/// Return the message signature or an error in case of failure.
 	pub fn sign(&self, public: &Public, message: &[u8]) -> Result<Signature, error::Error> {
-		let store = if let Some(store) = self.0.clone() {
-			store
-		} else {
-			return Err(error::Error::Keystore("no Keystore".to_string()));
-		};
+		let store = self
+			.0
+			.clone()
```

Comment:
I guess instead of cloning we could do `.as_ref()` ? We don't really need ownership of that `Arc`, do we?

Reply:
Using `.as_ref()` ultimately (after the `ok_or_else`) results in a `&Arc<dyn SyncCryptoStore>` instead of `Arc<dyn SyncCryptoStore>` which requires us to use the store as `&**store`. That looks really odd. 

Since `clone()` on an `Arc` is pretty cheap and idiomatic, I think that is clearer.  Although it may technically not really be needed.